### PR TITLE
Minor fix implicit cast CaresPTRResolver

### PR DESCRIPTION
**Pull request by jrdi**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

I'm getting the following errors when compiling master in OSX ARM64 (M1):

```bash
/private/tmp/ch_master/src/Common/CaresPTRResolver.cpp:131:94: error: implicit conversion loses integer precision: 'int64_t' (aka 'long long') to 'int' [-Werror,-Wshorten-64-to-32]
                number_of_fds_ready = poll(readable_sockets.data(), readable_sockets.size(), timeout);
                                      ~~~~                                                   ^~~~~~~
/private/tmp/ch_master/src/Common/CaresPTRResolver.cpp:131:86: error: implicit conversion loses integer precision: 'std::span<type-parameter-0-0, 18446744073709551615>::size_type' (aka 'unsigned long') to 'nfds_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
                number_of_fds_ready = poll(readable_sockets.data(), readable_sockets.size(), timeout);
                                      ~~~~                          ~~~~~~~~~~~~~~~~~^~~~~~
```

I'm not sure about what could be different with `timeout` since looks like c-ares `timeval` it's defined as `long` in all architectures. 

```c++
/*
 * Definition of timeval struct for platforms that don't have it.
 */

#ifndef HAVE_STRUCT_TIMEVAL
struct timeval {
 long tv_sec;
 long tv_usec;
};
#endif
```

But looks like `nfds_t` is defined as `unsigned int` in OSX ARM64 but as `unsigned long` in other architectures (this will explain the change from here also #42658)

### CODE REVIEW

**Comment by Algunenano** [comment]
Fast build [broken](https://s3.amazonaws.com/clickhouse-test-reports/42284/e77fcb0a99f2385191d04dce222f2e646d8bf51b/fast_test/runlog.log). It's broken in master -> https://github.com/ClickHouse/ClickHouse/pull/42747

File: src/Common/CaresPTRResolver.cpp [hunk 1]
```diff
@@ -128,7 +128,7 @@ namespace DB
             int number_of_fds_ready = 0;
             if (!readable_sockets.empty())
             {
-                number_of_fds_ready = poll(readable_sockets.data(), readable_sockets.size(), timeout);
+                number_of_fds_ready = poll(readable_sockets.data(), static_cast<unsigned int>(readable_sockets.size()), static_cast<int>(timeout));
```
Comment (azat) [inline src/Common/CaresPTRResolver.cpp:131]: Seems that `nfds_t` is `unsigned long int` (second argument), is it `unsigned int` for you?

Comment (azat) [reply src/Common/CaresPTRResolver.cpp:131]: I see, it is. Let's cast to `nfds_t` then (second argument).

```suggestion
                number_of_fds_ready = poll(readable_sockets.data(), static_cast<nfds_t int>(readable_sockets.size()), static_cast<int>(timeout));
```

**Comment by azat** [comment]
>I'm not sure about what could be different with timeout since looks like c-ares timeval it's defined as long in all architectures.

`poll` accept not `timeval` but simple `int`

**Comment by jrdi** [comment]
> > I'm not sure about what could be different with timeout since looks like c-ares timeval it's defined as long in all architectures.
> 
> `poll` accept not `timeval` but simple `int`

I know but `timetout` is computed by using `timeval`:

```
    int64_t CaresPTRResolver::calculate_timeout()
    {
        timeval tv;
        if (auto * tvp = ares_timeout(channel, nullptr, &tv))
        {
            auto timeout = tvp->tv_sec * 1000 + tvp->tv_usec / 1000;

            return timeout;
        }

        return 0;
    }
```

File: src/Common/CaresPTRResolver.cpp (hunk 1 above)
Comment (azat) [reply src/Common/CaresPTRResolver.cpp]: Can you try to rebase the branch? (So that CI sees labels)

**Comment by Algunenano** [comment]
The build is still broken in master. https://github.com/ClickHouse/ClickHouse/pull/42752 introduced only a partial fix.

From [the CI](https://s3.amazonaws.com/clickhouse-builds/42284/59686c37960df4c263a56cbf0100511d7b3b5930/binary_darwin_aarch64/build_log.log):

```
Oct 28 00:09:07 prlimit --as=10000000000 --data=5000000000 --cpu=1000 /usr/bin/ccache /usr/bin/clang++-15 --target=aarch64-apple-darwin -DAWS_SDK_VERSION_MAJOR=1 -DAWS_SDK_VERSION_MINOR=7 -DAWS_SDK_VERSION_PATCH=231 -DBOOST_ASIO_HAS_STD_INVOKE_RESULT=1 -DBOOST_ASIO_STANDALONE=1 -DCARES_STATICLIB -DENABLE_MULTITARGET_CODE=1 -DENABLE_OPENSSL_ENCRYPTION -DHAS_RESERVED_IDENTIFIER -DLZ4_DISABLE_DEPRECATE_WARNINGS=1 -DMZ_COMPAT_VERSION=110 -DPOCO_ENABLE_CPP11 -DPOCO_HAVE_FD_POLL -DPOCO_OS_FAMILY_UNIX -DUNALIGNED_OK -DWITH_COVERAGE=0 -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -I/build/build_docker/includes/configs -I/build/src -I/build/build_docker/src -I/build/build_docker/src/Core/include -I/build/base/widechar_width -I/build/base/base/.. -I/build/build_docker/base/base/.. -I/build/contrib/cctz/include -I/build/base/pcg-random/. -I/build/contrib/miniselect/include -I/build/contrib/zstd/lib -I/build/contrib/lz4/lib -I/build/contrib/minizip-ng -I/build/contrib/minizip-ng-cmake -isystem /build/contrib/libcxx/include -isystem /build/contrib/libcxxabi/include -isystem /build/contrib/jemalloc-cmake/include -isystem /build/contrib/cityhash102/include -isystem /build/contrib/boost -isystem /build/contrib/poco/Net/include -isystem /build/contrib/poco/Foundation/include -isystem /build/contrib/poco/NetSSL_OpenSSL/include -isystem /build/contrib/poco/Crypto/include -isystem /build/contrib/boringssl/include -isystem /build/contrib/poco/Util/include -isystem /build/contrib/poco/JSON/include -isystem /build/contrib/poco/XML/include -isystem /build/contrib/replxx/include -isystem /build/contrib/fmtlib-cmake/../fmtlib/include -isystem /build/contrib/magic_enum/include -isystem /build/contrib/double-conversion -isystem /build/contrib/dragonbox/include -isystem /build/contrib/re2 -isystem /build/build_docker/contrib/re2-cmake -isystem /build/contrib/zlib-ng -isystem /build/build_docker/contrib/zlib-ng-cmake -isystem /build/contrib/abseil-cpp -isystem /build/contrib/pdqsort -isystem /build/contrib/xz/src/liblzma/api -isystem /build/contrib/c-ares/src/lib -isystem /build/contrib/c-ares/include -isystem /build/contrib/c-ares-cmake/darwin -isystem /build/contrib/aws-c-common/include -isystem /build/contrib/aws-c-event-stream/include -isystem /build/contrib/aws/aws-cpp-sdk-s3/include -isystem /build/contrib/aws/aws-cpp-sdk-core/include -isystem /build/build_docker/contrib/aws-s3-cmake/include -isystem /build/contrib/brotli/c/include -isystem /build/contrib/snappy -isystem /build/build_docker/contrib/snappy-cmake -isystem /build/contrib/msgpack-c/include -isystem /build/contrib/fast_float/include -isystem /build/contrib/bzip2 -isystem /build/build_docker/contrib/bzip2 -std=c++20 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -march=armv8.2-a+simd+crypto+dotprod+ssbs -Xclang=-target-feature -Xclang=+ldapr -Wno-unused-command-line-argument -fasynchronous-unwind-tables -ffile-prefix-map=/build=. -falign-functions=32 -stdlib=libc++ -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -Wall -Wextra -Wframe-larger-than=65536 -Weverything -Wpedantic -Wno-zero-length-array -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c++20-compat -Wno-sign-conversion -Wno-implicit-int-conversion -Wno-implicit-int-float-conversion -Wno-ctad-maybe-unsupported -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-padded -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -O2 -g -DNDEBUG -O3 -g -gdwarf-4  -isysroot /build/cmake/darwin/../toolchain/darwin-aarch64 -mmacosx-version-min=10.15   -D OS_DARWIN -F/build/cmake/toolchain/darwin-aarch64/System/Library/Frameworks  -Werror -nostdinc++ -std=gnu++20 -Wno-reserved-identifier -MD -MT src/CMakeFiles/clickhouse_common_io.dir/Common/CaresPTRResolver.cpp.o -MF src/CMakeFiles/clickhouse_common_io.dir/Common/CaresPTRResolver.cpp.o.d -o src/CMakeFiles/clickhouse_common_io.dir/Common/CaresPTRResolver.cpp.o -c /build/src/Common/CaresPTRResolver.cpp
Oct 28 00:09:07 /build/src/Common/CaresPTRResolver.cpp:131:86: error: implicit conversion loses integer precision: 'std::span<type-parameter-0-0, 18446744073709551615>::size_type' (aka 'unsigned long') to 'nfds_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
Oct 28 00:09:07                 number_of_fds_ready = poll(readable_sockets.data(), readable_sockets.size(), static_cast<int>(timeout));
```

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
All builds are OK

**Pull request merged**